### DISCUSSION TITLE
S3 support for other than AWS, but compatible, repositories?

### DISCUSSION THREAD

**Discussion by glemoine62** [opengeos/leafmap#733] [Q&A] [answered]
I understand leafmap.s3_* functions (in common) are specific to AWS buckets. I would like to integrate data from a non-AWS repository which is fully compatible with the S3 standard. E.g. I can use boto3 to list_objects, etc. Since leafmap uses boto3 for S3 access, it should in principle be possible to use alternative repositories, e.g. by passing endpoint_url to the S3 session. Is this understanding correct?

**Comment by giswqs** [comment]
Yes, it should. It it doesn't work, please feel free to report an issue and provide a code snippet that can reproduce the issue

**Reply by glemoine62** [reply]
Do you happen to know how to pass in the endpoint_url? As an os.environ['ENDPOINT_URL'] (does not seem to work) or as one of the **kwargs?

**Reply by glemoine62** [reply]
Only the S3 functions that allow client_args to be passed work. I can do 

```
endpoint_properties = {}
endpoint_properties['endpoint_url'] = "https://url_of_alt_endpoint"
leafmap.s3_get_object('bucket_name', 'key', client_args=endpoint_properties)
```
but not ```leafmap.list_objects()``` which does not pass on any arguments to boto3.client() internally.

**Reply by giswqs** [reply, marked as answer]
This has been fixed in #734. Will release a new version later this week.
